boards/BMS/Inc/App/App_Soc.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "App_SharedExitCode.h"
+
+/**
+ * Given three state-of-charge (SoCs), check whether the difference between any
+ * two SoCs is less than or equal to the specified threshold. If any two SoCs
+ * pass the check, return the average value of those two SoCs.
+ * @param max_difference The maximum tolerable difference between any two given
```

Comment:
i'm not sure if thinking about this in terms of a max difference makes sense

if there's a single/double bit error in the EEPROM, or if a write is corrupted during power down, chances are the resulting float will be significantly different upon retrieval, and not within some range

the max difference doesn't hurt, but the way i did it in the past was just comparing the numbers, which should be equal. 

- if 3/3 numbers are equal, use that number
- if 2/3 numbers are equal, use that number and write over the third number
- if none are equal, we still need to run the car. we should map the voltage to an SoC by a LUT generated by the discharge data we gathered a while ago. it's rudimentary but a simple solution to an extremely unlikely problem

Reply:
I didn't want the rules to be so strict, and at worst we can just define the `max_abs_difference` to be `0`. 

As for the part about rational current limits, I agree but that's not the responsibility of this function. The caller would be using this function to see if none are equal, and handle it accordingly as you suggested.

Reply:
this is fine with me then, should i create an issue for doing the LUT?

Reply:
or actually do you think that what i recommended should be in the spec?

Reply:
I think you should just add it to the spec if possible